decoder/reference_targets.go

```diff
@@ -181,8 +181,17 @@ func (d *PathDecoder) decodeReferenceTargetsForBody(body hcl.Body, parentBlock *
 			}
 
 			if bSchema.Address.InferBody && bSchema.Body != nil {
+				if bSchema.Address.BodySelfRef {
```

Comment:
```suggestion
				var localAddr lang.Address
				if bSchema.Address.BodySelfRef {
```

I think this would avoid the need to update so many tests and reduce impact/risk for existing logic.